client/client.go

```diff
@@ -1066,3 +1083,7 @@ func addrsToUrls(addrs []string) []string {
 	}
 	return urls
 }
+
+func isMismatchLeader(err error) bool {
+	return strings.Contains(err.Error(), "mismatch leader id")
```

Comment:
I prefer not to use the string directly since we may change the error message later by accident.